### PR TITLE
Add State pattern to machine

### LINKED ISSUE #19: State design pattern for the machine

### PR DESCRIPTION
**Pull request by VincentNe**
Closes #19

### CODE REVIEW

File: src/main/java/euteam/cleanmachine/model/facility/MachineState.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package euteam.cleanmachine.model.facility;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public abstract class MachineState {
```
Comment (Raul6469) [inline src/main/java/euteam/cleanmachine/model/facility/MachineState.java]: Looks like this has to be an Interface: https://github.com/iluwatar/java-design-patterns/tree/master/state

Comment (VincentNe) [reply src/main/java/euteam/cleanmachine/model/facility/MachineState.java]: Can we change it to an interface? Last time we tried to put an interface into the db with jpa it didn't work.

Comment (Raul6469) [reply src/main/java/euteam/cleanmachine/model/facility/MachineState.java]: Oh yeah you're right, I think we have to keep it as a class to save it in the database... My bad

**Comment by Raul6469** [comment]
@VincentNe did you want to merge it? Because you just closed the pr

**Pull request merged**